geometry/cartesian_product.hpp

```diff
@@ -37,11 +37,23 @@ struct CartesianProductAdditiveGroup;
 template<typename Scalar,
          typename Tuple,
          typename = std::make_integer_sequence<int, std::tuple_size_v<Tuple>>>
-struct CartesianProductVectorSpace;
+class CartesianProductVectorSpace;
+
+// This class exports one function:
+//
+//   static constexpr ... Multiply(LTuple const& left, RTuple const& right);
+//
+// where the return types are appropriate for the operation.  The element types
+// of the tuples must have a structure of multiplicative ring.
```

Comment:
"The element types of the tuples must have a structure of multiplicative ring" is not a good description of the requirements (it sounds like the requirement for the product ring, which this is not).

Reply:
Done.